resolwe/storage/manager.py

```diff
@@ -28,6 +28,15 @@ def _lock_file_storage(self, file_storage_id: int) -> Optional[FileStorage]:
 
     def delete_single_location(self, file_storage: FileStorage, connector_name: str):
         """Delete given storage location."""
+        # Some other storage manager might have already processed this storage
+        # location before we did.
+        if not file_storage.storage_locations.filter(
+            connector_name=connector_name,
+            url=file_storage.subpath,
```

Comment:
Why do you need to filter by url, as it is already linked to the file storage?

Reply:
Fixed.